examples/window.py

```
@@ -0,0 +1,146 @@
+                except pygame.error:
+                    pass
+
+        if event.type == SHOW_WINDOW:
+            main_window.show()
+
+        if event.type == pygame.KEYDOWN:
+            if event.key == pygame.K_ESCAPE:
+                win, _ = windows_and_surfaces.pop()
+                win.destroy()
+
+                # Destroying the windows will not automatically do a QUIT on
+                # last window, unlike closing the windows manually.
+                if len(windows_and_surfaces) == 0:
+                    pygame.event.post(pygame.Event(pygame.QUIT))
+
+            if event.key == pygame.K_m:
+                main_window.maximize()
+
+            if event.key == pygame.K_n:
+                main_window.minimize()
+
+            if event.key == pygame.K_r:
+                main_window.restore()
+
+            if event.key == pygame.K_h:
+                main_window.hide()
+                pygame.time.set_timer(SHOW_WINDOW, 1000, 1)
+
+            if event.key == pygame.K_s:
+                win = pygame.Window("spawned window", (300, 300))
+                try:
+                    win.opacity = UNHOVERED_OPACITY
+                except pygame.error:
+                    pass
+                windows_and_surfaces.append((win, win.get_surface()))
+
+            if event.key == pygame.K_UP:
+                for win, _ in windows_and_surfaces:
+                    win.position += pygame.Vector2(0, -WIN_MOVE_SPEED)
+
+            if event.key == pygame.K_DOWN:
+                for win, _ in windows_and_surfaces:
+                    win.position += pygame.Vector2(0, WIN_MOVE_SPEED)
+
+            if event.key == pygame.K_LEFT:
+                for win, _ in windows_and_surfaces:
+                    win.position += pygame.Vector2(-WIN_MOVE_SPEED, 0)
+
+            if event.key == pygame.K_RIGHT:
+                for win, _ in windows_and_surfaces:
+                    win.position += pygame.Vector2(WIN_MOVE_SPEED, 0)
```

## API comment

SMH my head, `window.position` *is also broken under direct wayland*

Nothing moves when I press the arrow keys. :cry: 

All these "direct wayland" issues are fine/ignorable for now because these issues don't happen under XWayland which SDL2 picks by default, but this could be a problem in the SDL3 era if SDL3 devs decide to do wayland-default.

---

This is by design. Wayland does not want applications to know this for API design and security reasons. There is nothing SDL3 can do about it short of sending patches to KDE and GNOME and sway

---

Do I need to add try/except pygame.error blocks around them then? I did it for opacity given your previous report.

I'm also unsure if we really want to raise an exception if the operation is not supported. We could return True/False about success?

---

In this case the position is silently ignored by SDL, like `always_on_top`. Only opacity explicitly errors.

I think for this example, we should not have this position-moving thing as a highlight, and make it more low-key somehow?